gap/attr.gi

```diff
@@ -700,6 +700,52 @@ end);
 # returns the vertices (i.e. numbers) of <D> ordered so that there are no
 # edges from <out[j]> to <out[i]> for all <i> greater than <j>.
 
+InstallMethod(UnweightedBellmanFord, "for a digraph by out-neighbours",
+[IsDigraph, IsPosInt],
+function(digraph, source)
+  local distance, n, predecessor, i, inf, u, v, edge, w;
+  n := DigraphNrVertices(digraph);
+  # wouldn't  work for weighted digraphs
+  inf := n + 1;
+  distance := List([1 .. n], x -> 0);
+  predecessor := List([1 .. n], x -> 0);
+  for i in DigraphVertices(digraph) do
```

Comment:
This loop is unnecessary - you should just do `distance := List([1 .. n], x -> inf)` to initialise the `distances` list to contain the value `inf`, and the same for `predecessor` (which already contains all `0`s from the line above).

Better would be to use the descriptive function `ListWithIdenticalEntries(n, inf)`.